lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart

```diff
@@ -183,6 +192,106 @@ class RecipientStepState extends State<RecipientStep>
     );
   }
 
+  void _setTimeLock() async {
+    final DateTime? value = await showTimelockPicker(context: context);
+    if (value == null) {
+      setState(() {
+        calendarValue = null;
+        timelockSet = false;
+      });
+    } else {
+      setState(() {
+        calendarValue = value;
+        timelockSet = true;
+        _currIndex = _currIndex == 0 ? 1 : 0;
+      });
+    }
+  }
+
+  void _clearTimeLock() async {
+    setState(() {
+      calendarValue = null;
+      timelockSet = false;
+      _currIndex = _currIndex == 0 ? 1 : 0;
+    });
+  }
+
+  String _formatTimeLock() =>
+      DateFormat('h:mm a E, MMM dd yyyy ').format(calendarValue!);
+
+  _buildCalendarDialogButton(BuildContext context) {
+    final theme = Theme.of(context);
+    Column timelockWidget = Column(
+      children: [
+        SizedBox(height: 15),
+        Row(
+            mainAxisAlignment: MainAxisAlignment.start,
```

Comment:
Can we add a tooltip saying 'The recipient will not be able to spend the coins before this date and time'?